[Mok Yao Hui] iP

laxus2308 opened the pull request:
# ~~Anya~~Wanya

> "Nothing Mama Teaches Me Ever Helps." - Anya Forger [(source)](https://animegalaxyofficial.com/20-best-spy-x-family-quotes-by-anya-loid-and-yor-forger/)

***Warning!!***  Wanya isn't a mind reader like Anya, but it frees your mind of having to remember things you need to do!

It's,
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/laxus2308/ip/releases/tag/A-Jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉 👍 

And it is **FREEEE!**

Features:
- [X] Managing tasks
- [X] Find your tasks easily
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)


## 
If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```
 public static void main(String[] args) {
        Wanya wanya = new Wanya("tasks.txt");
        wanya.run();
 }
```

Commit:
Add Gradle support

laxus2308 committed:
Added Greet, Echo and Exit

laxus2308 committed:
able to add task and list tasks

laxus2308 committed:
Added mark and unmark tasks

laxus2308 committed:
Added todo, deadline, event

laxus2308 committed:
Added automated Text UI Testing

laxus2308 committed:
Added exceptions to handle error

laxus2308 committed:
Added delete and handle exceptions for delete

laxus2308 committed:
Add ability to save and auto load from storage

laxus2308 committed:
Update data storage

Storing of task description gets pushed by 1 spacing
each time the bot is closed.
Fixed by removing white spaces when saving to file.
Removed print statements used for testing.

laxus2308 committed:
Add DateTimeParser for Wanya to understand date and time

DateTimeParser enable Wanya to recognise String of format
"yyyy-MM-dd HH:mm" as dates and time respectively.

laxus2308 committed:
Add exceptions and test for DateTimeParser

Add checks to ensure program will not crash if date is not provided
or format of date is invalid. Remind users the correct format
required to add deadlines and events.

Conduct automated text-ui testing for regression testing.

laxus2308 committed:
Dummy Commit

Made some unecessary changes to master branch to be 1 commit ahead.

laxus2308 committed:
Merge branch 'branch-level-7'

laxus2308 committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java

laxus2308 committed:
Fix bug of missing details from Wanya file

Reason: I should have changed the file name from Duke to Wanya
from the master branch before even creating the new branches, which
led to the changes made in the Level-8 branch to not be reflected in
the master Branch.

laxus2308 committed:
Fix bug of date time format not displayed properly

Initially, the date time format shown on the deadline and event
are in the DateTime format rather than my intended format. I have
missed out the function call when converting toString method.
Code is working as intended and conducted text-UI test.

laxus2308 committed:
Refactor code for more OOP

Create 4 new classes
- Ui to deal with interaction with users
- Parser to make sense of the command user inputs
- TaskList to contain task list and handle tasks operations
- Storage to load and save task list

laxus2308 committed:
Add in packages

- Divide classes into packages
- Reformat tasks string format in Storage

laxus2308 committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

laxus2308 committed:
Add gradle support

laxus2308 committed:
Edit Parser ability to check for empty task

Tasks that have empty description such as "todo   "
is recognized as a valid task by the parser, even though
it should not be valid.
Trim function removes white spaces to fix the issue.

laxus2308 committed:
Add JUnit tests

Add JUnit tests for DateTimeParser, Deadline, Event and Todo
Checks:
- parsing of date
- parsing of date with time
- parsing of incorrect date
- parsing of correct date with incorrect time

- deadline String representation
- deadline Storage String

- Event String representation
- Event Storage String

- ToDo String representation
- ToDo Storage String

laxus2308 committed:
Add jar file

laxus2308 committed:
Add JavaDoc

Added JavaDoc comments to public methods and public classes

laxus2308 committed:
Add comments at referenced codes

laxus2308 committed:
Change method names for setter methods of boolean variables

laxus2308 committed:
Add in @@author for reused code

laxus2308 committed:
Add find function

Users can now search for related tasks in task list with
a given keyword.

laxus2308 committed:
Remove unecessary "this" from code

laxus2308 committed:
Merge branch 'branch-A-JavaDoc'

laxus2308 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/wanya/task/Task.java

laxus2308 committed:
Merge branch 'branch-Level-9'

laxus2308 committed:
Update find function to act differently if no match was found.

Initially, when no match of keyword from current tasks in task list,
Wanya will still print "Here are the matching tasks:".

Wanya should not print that if there are no matching tasks.

Wanya now informs the user there are no matching tasks if there
are no matching tasks.

eugenetayyj commented on src/main/java/wanya/parser/Parser.java:
```diff
@@ -0,0 +1,109 @@
+     */
+    public static int checkTaskNumber(String[] inputs, TaskList tasks) throws WanyaException {
+        //handle error without task number
+        if (inputs.length == 1 || inputs[1].trim().equals("")) {
```
should input[1] be extracted out and be referenced as a variable instead? Since you used it a few times after.

eugenetayyj commented on src/main/java/wanya/parser/Parser.java:
```diff
@@ -0,0 +1,109 @@
+                tasks.showTasks();
+            } else if (command.equals("mark")) {
+                int indexToMark = checkTaskNumber(inputs, tasks);
+                tasks.get(indexToMark - 1).setComplete();
```
Would it be better to extract out these lines of code into a command and let the command handle this instead? For example, returning a MarkCommand Object and allow the MarkCommand Object to .setComplete().

eugenetayyj commented on src/main/java/wanya/parser/Parser.java:
```diff
@@ -0,0 +1,109 @@
+                checkTask(inputs, command);
+                try {
+                    tasks.addEvent(inputs[1]);
+                } catch (DateTimeException e) {
```
Perhaps it could be better to allow the function as a whole to catch the DateTimeException rather than at each task?

eugenetayyj reviewed (commented):
Overall, I like the code quality. However maybe some lines of code can be extracted out to make the code more dynamic and for the function to be shorter. Overall well done!👍🏽

laxus2308 committed:
Add GUI to replace text based interface

Features implemented:
1. Add GUI (mainWindow, dialogBox and pictures)
2. Refactor functions method to return String instead of print

laxus2308 committed:
Merge branch 'Level-10'

laxus2308 committed:
Add checkstyle

laxus2308 committed:
Merge branch 'A-CheckStyle'

leehuiyulaura commented on src/main/java/wanya/Launcher.java:
```diff
@@ -0,0 +1,11 @@
+package wanya;
```
I like that you have created and neatly organised all your classes into a few packages as it makes it easy for readers to understand the purpose of each of your classes.

leehuiyulaura commented on src/main/java/wanya/Storage.java:
```diff
@@ -0,0 +1,60 @@
+import java.util.List;
+
+// Below code is adapted from
+// https://github.com/se-edu/addressbook-level2/blob/master/src/seedu/addressbook/storage/StorageFile.java
```
I like that you reference external websites and sources that you use to help you with your code.

leehuiyulaura commented on src/test/java/wanya/ToDoTest.java:
```diff
@@ -0,0 +1,26 @@
+    }
+
+    @Test
+    public void todoToStorageTest() {
```
I would like to commend you on how you have strictly followed the Java coding standard of using Egyptian, K&R style brackets throughout your entire code.

leehuiyulaura commented on src/main/java/wanya/task/TaskList.java:
```diff
@@ -0,0 +1,219 @@
+    /**
+     * Finds a list of tasks that contains the keyword and print it.
+     *
+     * @param keyword word to find matching tasks.
```
Perhaps you would like to capitalise the first letter in the word "word"? As I believe that this may violate JavaDoc standards.

leehuiyulaura commented on src/main/java/wanya/task/Task.java:
```diff
@@ -0,0 +1,94 @@
+    /**
+     * Creates a task when given a task name and completeness.
+     *
+     * @param taskName name of the task.
```
I believe that you should capitalise the first letter of the word "name" in the @param line to adhere to Javadoc coding standards. I believe that there were other parts of the code in which you were unable to do so as well and I think it would be good to fix it just to be safe!

leehuiyulaura commented on src/main/java/wanya/task/Event.java:
```diff
@@ -0,0 +1,60 @@
+package wanya.task;
+
+import wanya.parser.DateTimeParser;
```
I like that you added a line break to separate the java imports and package imports as it makes it clearer and easy to reference!

leehuiyulaura reviewed (commented):
Overall, your code is extremely comprehensive, easy to understand and mostly strictly adheres to the Java coding standards and conventions. Job well done!

laxus2308 committed:
Add Assertions to Parser and TaskList

Include Assertions in current methods to verify underlying
assumptions in code to ensure mistakes made by the programmer
will not cause the entire bot to malfunction, especially when
loading saved files that may be corrupted.

laxus2308 committed:
Refactor code to improve code quality

1. For methods that were previously too long, additional methods
were created to replace portions of the code within the if else
blocks to allow easier debugging in the future by isolating the method
that is causing the bug.

2. Avoid complicated expression in if statements. I calculated the
intermediate values separately and include the final value for the
condition in if statement.

3. Get rid of 'magic' string that are returned from functions. Add
a variable name to the String to self explain what does the String
represent before returning it.

laxus2308 committed:
Merge pull request #1 from laxus2308/branch-A-Assertions

Add Assertions to Parser and TaskList

laxus2308 committed:
Merge pull request #2 from laxus2308/branch-A-CodeQuality

Refactor code to improve code quality

laxus2308 committed:
Add missing semicolon

laxus2308 committed:
Add Continuous Integration(CI)

laxus2308 committed:
Add DoWithinPeriod tasks

laxus2308 committed:
Merge branch 'B-DoWithinPeriodTasks'

laxus2308 committed:
Add files via upload

laxus2308 committed:
Update README.md

laxus2308 committed:
Update README on doc

laxus2308 committed:
Edit avatar pictures

laxus2308 committed:
Update Readme
